### PR TITLE
Fix #1228 - Move exe patching to loader

### LINKED ISSUE #1228: Move exe patching to loader
**Is your feature request related to a problem? Please describe.**

If a user install GTA after MTA, MTA will not patch their GTA exe and will fail to start.

**Describe the solution you'd like**

Move (or copy) exe patching from the installer and to the loader.

**Describe alternatives you've considered**

**Additional context**

We have received a lot of support messages relating to this. It would be better to instead just attempt to patch on failure instead of presenting some error message.

Current source is here: https://github.com/multitheftauto/mtasa-blue/blob/master/Shared/installer/nightly.nsi

[This line](https://github.com/multitheftauto/mtasa-blue/blob/master/Shared/installer/nightly.nsi#L1212) refers to "vpatch", which might be this: https://www.tibed.net/vpatch/

### PR DESCRIPTION
**Pull request by ccw808**
fixes #1228

### CODE REVIEW

**Comment by qaisjp** [comment]
Haven't read through the code properly yet, but I noticed that the original patcher code is still there - is that intentional? 

Thought about fetching the patch from a server instead of shipping it with every update? (Granted, it is only 4.5 MB.)

File: Client/loader/CInstallManager.cpp [hunk 1]
```diff
@@ -527,6 +542,81 @@ void MigrateFile(const SString& strFilenameOld, const SString& strFilenameNew)
...
+        return "ok";
+    }
+
+    // Check required GTA file is correct
+    if (GenerateHashHexStringFromFile(EHashFunctionType::MD5, strPatchBase) != strPatchBaseMd5)
+    {
+        AddReportLog(5053, SString("_ProcessGtaVersionCheck: Incorrect file '%s'", *strPatchBase));
+        SString strMessage(_("MTA:SA cannot continue because the following files are incorrect:"));
+        strMessage += "\n\n" + strPatchBase;
+        BrowseToSolution("gengta_pakfiles", ASK_GO_ONLINE, strMessage);
```
Comment (qaisjp) [inline Client/loader/CInstallManager.cpp:582]: I'm guessing https://updatesa.mtasa.com/sa/trouble/?tr=gengta_pakfiles and the one below will be created when this is merged?

**Review by qaisjp — APPROVED** [review]
LGTM

**Comment by ccw808** [comment]
> Haven't read through the code properly yet, but I noticed that the original patcher code is still there - is that intentional?

Yes
 
> Thought about fetching the patch from a server instead of shipping it with every update? (Granted, it is only 4.5 MB.)

Reasons for not downloading are: 1) Launcher currently doesn't depend on internet access, so there might be firewall issues and 2) data files are only in the full update. Slim update is used if client already has certain files.

File: Client/loader/CInstallManager.cpp (hunk 1 above)
Comment (ccw808) [reply Client/loader/CInstallManager.cpp:582]: Correct

**Pull request merged**